src/drivers/src/encoder_odometry_node.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: March 29th, 2018
+ * Description: A node the that translates the encoder readings (published in a
+ *              JointState message as # of ticks) to a ROS Odometry message
+ *              Originally designed for use with the Phidget 1047 encoder
+ */
+
+#include <EncoderOdometryNode.h>
+
+
+int main(int argc, char **argv){
+    std::string node_name = "encoder_to_odometry";
+
+    // Setup our Encoder -> Odometry conversion node
+    EncoderOdometryNode encoder_odometry_node(argc, argv, node_name);
+
+    // The rate at which to publish estimated Odometry messages (in hz).
+    // We use this because we want to make velocity predictions based off
+    // several sets of encoder readings (to allow for averaging and reduce
+    // estimation error)
+    int odometry_estimate_refresh_rate;
+
+    // Get the desired refresh rate of the Odometry message estimation
+    ros::NodeHandle private_nh("~");
+    SB_getParam(private_nh, "odom_msg_refresh_rate", odometry_estimate_refresh_rate, 10);
+
+    // Set the refresh rate to the desired odom estimation rate
+    ros::Rate loop_rate(odometry_estimate_refresh_rate);
+
+    // Keep making estimates until we're told to stop
+    while (ros::ok()){
+        encoder_odometry_node.publishEstimatedOdomMsg();
+        ros::spinOnce();
+        loop_rate.sleep();
+    }
```

Comment:
Could consider using a [Timer](http://wiki.ros.org/roscpp/Overview/Timers) within the node itself perhaps?

Reply:
Huh, didn't know this was a thing, TIL! Anyhow, my initial thought was that it might be a bit harder to follow then the current structure (using `while(ros::ok())`) but after some more consideration this seems like a much better solution. People on the team are already used to having basically all the functionality in the class itself, and are already used to the concept of callback functions, so this isn't much of a leap in understanding. Changed!